### DISCUSSION TITLE
Environment variable in vercel not working

### DISCUSSION THREAD

**Discussion by devmahmud** [vercel/next.js#60819] [Help]
### Summary

I am trying to validate the env variables like this, which is working on local machine perfectly. But when building on vercel, getting an error:

Here is the schema:

```tsx
import { z } from "zod";

const envSchema = z.object({
	API_ENDPOINT: z.string(),
	GOOGLE_OAUTH_CLIENT_ID: z.string(),
});

export const env = envSchema.parse(process.env);
```
![image](https://github.com/vercel/next.js/assets/19981097/649d52c3-3d3f-4229-9e61-ee8e7de056a3)

Here is the error I am getting:
![image](https://github.com/vercel/next.js/assets/19981097/4f92cd9f-7745-4cb3-ad6c-ac6f31ff797f)

**Comment by icyJoseph** [comment]
Hi,

Have you added them to the Vercel dashboard?

**Reply by devmahmud** [reply]
Yes, I have attached the screenshot above. Please check.

**Comment by ezeparziale** [comment]
Check if the env var is available in the environments.

![Captura de pantalla 2024-01-18 a la(s) 15 25 38](https://github.com/vercel/next.js/assets/82482391/ae379c3a-3e5e-4730-a90c-a583c9c73e6a)

And the redeploy with no cache

**Comment by devmahmud** [comment]
I was able to fix it by explicitly adding another preview env variable. Cause another preview is targeted on staging branch.

![image](https://github.com/vercel/next.js/assets/19981097/a4529937-bda5-46e7-a10b-d4ffc85f3fca)

Thanks everyone for your support.
